WordPress/openverse-frontend
# Set up global audio player

## Fixes


Fixes #589 by @zackkrida 

## Description


This PR
- creates the global audio section as a part of the layout
  - this section renders the audio track in the 'Global' layout for mobile screens
- removes the defunct `VAudioController` component (as there is no longer `<audio>` in it)
  - those duties have been migrated to `VAudioTrack` itself
- cleans up and refactors code to keep the flow of control between `VPlayPause`, `VWaveform`, `VAudioTrack`, Vuex active media store and `<audio>` sane
- adds more sample music data to make testing pleasant 

## Testing Instructions

0. Checkout the branch and run the dev server.
1. Resize the screen to a mobile layout using dev tools, bonus points for testing on an actual phone.
2. Play a song. The global audio layout should appear and work.
3. Change it midway. The new song should immediately play and the play-pause icons should be correct for both songs.
4. Pause the song and play again. It should pick up where it paused.
5. Pause from the global layout and play from the search result icon and vice versa. It should work.
6. Close the player when the song is playing. It should pause and the global layout should disappear.
7. Keep the API running on port 8000 for cover art. Images should show up in the right places.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
